addons/project/wizard/project_share_wizard.py

```diff
@@ -140,6 +151,16 @@ def action_share_record(self):
             return
         on_invite = self.env['res.users']._get_signup_invitation_scope() == 'b2b'
         new_portal_user = self.collaborator_ids.filtered(lambda c: c.send_invitation and not c.partner_id.user_ids) and on_invite
+        if self.partners_no_email:
+            return {
+                'name': _('No Email Address for Some Partners'),
+                'view_mode': 'form',
+                'res_model': 'project.share.wizard',
+                'views': [(self.env.ref('project.partner_no_email_list_wizard').id, 'form')],
+                'type': 'ir.actions.act_window',
+                'res_id': self.id,
+                'target': 'new',
+            }
```

Comment:
There seems to be code duplication. I think we should use project_share_wizard_action